src/App/Fossa/IAT/Types.hs

```diff
@@ -0,0 +1,34 @@
+{-# LANGUAGE DeriveDataTypeable #-}
+
+module App.Fossa.IAT.Types (
+  Fingerprint (..),
+  Raw,
+  UserDefinedBinaryAssertion (..),
+) where
+
+import Data.Aeson
+import Data.Text
+import Data.Typeable
+
+-- | Fingerprint uniquely idenfies a file, derived from its content.
+-- Fingerprints are backed by base16 representations of underlying data.
+newtype Fingerprint t = Fingerprint Text
+
+instance ToJSON (Fingerprint t) where
+  toJSON = toJSON . toText
+
+toText :: Fingerprint t -> Text
+toText (Fingerprint x) = x
+
+-- | Raw describes a fingerprint calculated from the full unmodified content of a file.
+-- Raw fingerprints are implemented as SHA256 hashes.
+data Raw deriving (Typeable)
```

Comment:
I'm not sure what the purpose of this type is.  I've seen it used in `Fingerprint Raw`, but it's not clear to me why we're using this type

Reply:
My intent was to support other kinds of fingerprints if we want them in the future. I'm happy to avoid the extra complexity for now though, upon thinking on it more.